doc/language_guide.md

```diff
@@ -386,6 +386,69 @@ INTO evaluate_result_table;
 - *attr_expr* sets attributes when doing evaluation. You can set `validation.metrics` to indicate which metrics will be outputed to the result table, e.g. `validation.metrics="Accuracy,AUC"`, you can find more supported metric names for Keras models [here](https://www.tensorflow.org/api_docs/python/tf/keras/metrics) and XGBoost models [here](https://xgboost.readthedocs.io/en/latest/parameter.html).
 - *evaluate_result_table* is the result table that stores the evaluation results.
 
+## Optimization Syntax
+
+SQLFlow uses the `TO MAXIMIZE` and `TO MINIMIZE` clauses to describe and solve the math programming problems. 
+
+A SQLFlow `TO MAXIMIZE` or `TO MINIMIZE` clause consists of select, objective, constraints, attributes, solver and into expressions. 
+
+```sql
+SELECT select_expr [, select_expr ...]
+FROM table_references
+  [WHERE where_condition]
+  [LIMIT row_count]
+TO MAXIMIZE|MINIMIZE
+  objective_expr
+CONSTRAINT
+  constraint_expr [GROUP BY column_name]
+  [, constraint_expr [GROUP BY column_name]...]
+WITH
+  variables="variable_value(column_name, ...)"
+  var_type="Integers|Reals|Binary|NonNegativeIntegers|..."
+[USING solver_name]
+INTO result_table_name;
+```
+
+A real world SQL statement for optimization is as follows:
+
+```sql
+SELECT c1, c2, c3 FROM my_db.my_table
+TO MAXIMIZE SUM(x * c1)
+CONSTRAINT
+    SUM(x) <= c2 GROUP BY c1
+    x <= 3
+WITH
+    variables="x(c3)",
+    var_type="NonNegativeIntegers"
+USING glpk
+INTO my_db.my_result_table;
+```
+
+The objective expression following by `TO MAXIMIZE` means that we want to maximize the value of `SUM(x * c1)`. 
```

Comment:
1. use a list to describe the components in the statement
2. The Objective expression -> The `objective_expr`

Reply:
Done.